mesonbuild/mtest.py

```diff
@@ -1822,7 +1823,7 @@ def doit(self) -> int:
             os.chdir(self.options.wd)
             runners: T.List[SingleTestRunner] = []
             for i in range(self.options.repeat):
-                runners.extend(self.get_test_runner(test) for test in tests)
+                runners.extend(self.get_test_runner(test, i + 1) for test in tests)
```

Comment:
Python usually uses indexes starting at 0. I think I would prefer to see the increment when populating the env var, instead of here. In my opinion this would make the code easier to follow.